File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    answer = True
    arr=[]
    
    for parent in s:
        if parent=="(":
            arr.append(parent)
        if parent==")":
            if len(arr)==0:
                answer=False
                break
            else:
                arr.pop()
    if len(arr)>0:
        answer=False

    return answer
        
            
    

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.java
import java.util.*;
import java.io.*;
class Solution {
    boolean solution(String s) {
        boolean answer = true;
        Stack<String> stack = new Stack<>();
        for(char a:s.toCharArray()){
            if(a=='('){
                stack.push(String.valueOf(a));
            }else{
                if (stack.isEmpty()){
                    return false;
                }else{
                    stack.pop();
                }
            }
        }
        

        return stack.isEmpty();
    }
}